typescript/cli/src/deploy/utils.ts

```diff
@@ -55,13 +56,20 @@ export async function runPreflightChecksForChains({
   assertSigner(signer);
   logGreen('✅ Signer is valid');
 
-  const sufficient = await gasBalancesAreSufficient(
-    multiProvider,
-    signer,
-    chainsToGasCheck ?? chains,
-    minGas,
-  );
-  if (sufficient) logGreen('✅ Balances are sufficient');
+  const allSufficient = (
+    await nativeBalancesAreSufficient(
+      multiProvider,
+      signer,
+      chainsToGasCheck ?? chains,
+      minGas,
+    )
+  ).every((sufficient) => sufficient);
+
+  if (allSufficient) logGreen('✅ Balances are sufficient');
+  else
+    logRed(
```

Comment:
Like mentioned offline, wouldn't do this. If you get the warning and then top up later, we will still show this warning. What actually makes more sense to me, is to fetch all the imbalances all at once, list those, and then ask the user once if they want to continue